main.go

```diff
@@ -349,6 +356,87 @@ func _render(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, sketc
 	return svg, nil
 }
 
+func renderPDF(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, sketch bool, pad int64, outputPath string, page playwright.Page, ruler *textmeasure.Ruler, diagram *d2target.Diagram, pdf *pdflib.GoFPDF, boardPath []string, isScenario bool) (svg []byte, err error) {
+	var isRoot bool
+	if pdf == nil {
+		pdf = pdflib.Init()
+		isRoot = true
+	}
+
+	var currBoardPath []string
+	// Root board doesn't have a name, so we use the output filename
+	if diagram.Name == "" {
+		ext := filepath.Ext(outputPath)
+		trimmedPath := strings.TrimSuffix(outputPath, ext)
+		splitPath := strings.Split(trimmedPath, "/")
+		rootName := splitPath[len(splitPath)-1]
+		currBoardPath = append(boardPath, rootName)
+	} else {
+		currBoardPath = append(boardPath, diagram.Name)
+	}
+
+	if !isScenario {
```

Comment:
In Terrastruct, Scenarios are folders that can only contain steps, so I chose to ignore the generation step for scenarios. 

However, now we can have scenarios that are a diagram? If that's the case what's the point of having scenarios vs layers?

Reply:
> However, now we can have scenarios that are a diagram? If that's the case what's the point of having scenarios vs layers?

Code reuse. Scenarios are layers where the base of the layer is another diagram. So you have all this code creating a diagram and then a scenario of said diagram where you change just a few things.

Reply:
If a scenario exists but only contains steps, is the expectation that it has a "page" within the pdf?

Reply:
Maybe like a section heading. Though I don't think right now we have a way to detect that which is unfortunate.

Even for general output, if a scenario only contains steps and no modifications of its own, it shouldn't render.